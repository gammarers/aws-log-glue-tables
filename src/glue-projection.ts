
export namespace GlueProjection {

  export interface TimestampRange {
    readonly start: string;
    readonly end?: string;
  }

  export interface Timestamp {
    readonly range: TimestampRange;
  }

}