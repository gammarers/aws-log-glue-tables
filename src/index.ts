export * from './tables/bucket-static-web-hosting-access-log-table';
export * from './tables/cloudfront-access-log-table';
export * from './glue-projection';
