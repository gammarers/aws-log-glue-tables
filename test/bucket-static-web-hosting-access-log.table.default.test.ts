import { App, Stack } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as glue from 'aws-cdk-lib/aws-glue';
import * as s3 from 'aws-cdk-lib/aws-s3';
import { BucketStaticWebHostingAccessLogTable } from '../src';

describe('BucketStaticWebHostingAccessLogTable default Testing', () => {

  const app = new App();
  const stack = new Stack(app, 'TestingStack');

  const bucket = new s3.Bucket(stack, 'LogBucket');

  const databaseName = 'example-database';

  const tableName = 'example-bucket-access-log-table';

  const table = new BucketStaticWebHostingAccessLogTable(stack, 'BucketStaticWebHostingAccessLogTable', {
    logBucket: bucket,
    glueDatabaseName: databaseName,
    glueTableName: tableName,
    location: 'path/to/location',
    timestampRage: {
      start: '2024/11/01',
    },
  });

  it('Is Glue Table', () => {
    expect(table.glueTable).toBeInstanceOf(glue.CfnTable);
  });

  const template = Template.fromStack(stack);

  it('Should match snapshot', () => {
    expect(template.toJSON()).toMatchSnapshot();
  });
});