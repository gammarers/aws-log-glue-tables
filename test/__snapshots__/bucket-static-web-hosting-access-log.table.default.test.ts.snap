// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BucketStaticWebHostingAccessLogTable default Testing Should match snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "BucketStaticWebHostingAccessLogTableAccessLogGlueTableF05403E6": {
      "Properties": {
        "CatalogId": {
          "Ref": "AWS::AccountId",
        },
        "DatabaseName": "example-database",
        "TableInput": {
          "Name": "example-bucket-access-log-table",
          "Parameters": {
            "projection.enabled": "true",
            "projection.timestamp.format": "yyyy/MM/dd",
            "projection.timestamp.range": "2024/11/01,NOW",
            "projection.timestamp.type": "date",
            "storage.location.template": {
              "Fn::Join": [
                "",
                [
                  "s3://",
                  {
                    "Ref": "LogBucketCC3B17E8",
                  },
                  "/path/to/location/\${timestamp}/",
                ],
              ],
            },
          },
          "PartitionKeys": [
            {
              "Name": "timestamp",
              "Type": "string",
            },
          ],
          "StorageDescriptor": {
            "Columns": [
              {
                "Name": "bucket_owner",
                "Type": "string",
              },
              {
                "Name": "bucket_name",
                "Type": "string",
              },
              {
                "Name": "request_date_time",
                "Type": "string",
              },
              {
                "Name": "remote_ip",
                "Type": "string",
              },
              {
                "Name": "requester",
                "Type": "string",
              },
              {
                "Name": "request_id",
                "Type": "string",
              },
              {
                "Name": "operation",
                "Type": "string",
              },
              {
                "Name": "key",
                "Type": "string",
              },
              {
                "Name": "request_uri",
                "Type": "string",
              },
              {
                "Name": "http_status",
                "Type": "string",
              },
              {
                "Name": "errorcode",
                "Type": "string",
              },
              {
                "Name": "bytessent",
                "Type": "bigint",
              },
              {
                "Name": "object_size",
                "Type": "bigint",
              },
              {
                "Name": "total_time",
                "Type": "string",
              },
              {
                "Name": "turn_around_time",
                "Type": "string",
              },
              {
                "Name": "referrer",
                "Type": "string",
              },
              {
                "Name": "user_agent",
                "Type": "string",
              },
              {
                "Name": "version_id",
                "Type": "string",
              },
            ],
            "InputFormat": "org.apache.hadoop.mapred.TextInputFormat",
            "Location": {
              "Fn::Join": [
                "",
                [
                  "s3://",
                  {
                    "Ref": "LogBucketCC3B17E8",
                  },
                  "/path/to/location/",
                ],
              ],
            },
            "OutputFormat": "org.apache.hadoop.hive.ql.io.HiveIgnoreKeyTextOutputFormat",
            "SerdeInfo": {
              "Parameters": {
                "input.regex": "([^ ]*) ([^ ]*) \\[(.*?)\\] ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ("[^"]*"|-) (-|[0-9]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ("[^"]*"|-) ([^ ]*)(?: ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*) ([^ ]*))?.*$",
              },
              "SerializationLibrary": "org.apache.hadoop.hive.serde2.RegexSerDe",
            },
          },
          "TableType": "EXTERNAL_TABLE",
        },
      },
      "Type": "AWS::Glue::Table",
    },
    "LogBucketCC3B17E8": {
      "DeletionPolicy": "Retain",
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
